scenario_player/utils/configuration/token.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+import pathlib
+import uuid
+from typing import Optional, Union
+
+import structlog
+
+from scenario_player.exceptions.config import TokenConfigurationError
+from scenario_player.utils.configuration.base import ConfigMapping
+
+log = structlog.get_logger(__name__)
+
+
+class TokenConfig(ConfigMapping):
+    def __init__(self, loaded_yaml: dict, data_path: pathlib.Path):
+        super(TokenConfig, self).__init__(loaded_yaml.get("token", {}))
+        self._token_id = uuid.uuid4()
+        self._name = None
+        self._token_file = data_path.joinpath("token.info")
+        self.validate()
+
+    @staticmethod
+    def assert_option(expression, err: Optional[Union[str, Exception]] = None):
+        """Assert the given expression and raise a NodeConfigurationError if it fails."""
+        if isinstance(err, str):
+            err = TokenConfigurationError(err)
+        return ConfigMapping.assert_option(expression, err)
+
+    def validate(self):
+        """Validate the configuration section.
+
+        Asserts that:
+
+            * Mutually exclusive options are not truthy
+        """
+        mutual_exclusive_ops = ("address", "reuse")
+        if all(option in self.dict for option in mutual_exclusive_ops):
+            self.assert_option(
+                bool(self["address"]) != self["reuse"],
+                f"Token settings {mutual_exclusive_ops} are mutually exclusive.",
+            )
+
+    @property
+    def name(self):
+        if not self._name:
```

Comment:
If reuse is true the name must be read from the reused contract.

Reply:
I would expect a `token_name` key to be present in the `token.info` file, that way I do not have to talk to the nodes at setup. 

This of course requires the step of saving token information to be extended accordingly.